src/psyclone/tests/lfric_ref_elem_test.py

```diff
@@ -277,5 +280,166 @@ def test_union_refelem_gen(tmpdir):
+                      ignore_comments=False)
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    gen = str(kernel.gen_stub)
+
+    output = (
+        "  MODULE testkern_ref_elem_mod\n"
+        "    IMPLICIT NONE\n"
+        "    CONTAINS\n"
+        "    SUBROUTINE testkern_ref_elem_code(nlayers, rscalar_1, "
+        "field_2_w1, field_3_w2, field_4_w2, field_5_w3, ndf_w1, undf_w1, "
+        "map_w1, ndf_w2, undf_w2, map_w2, ndf_w3, undf_w3, map_w3, "
+        "nfaces_re_h, nfaces_re_v, normals_to_horiz_faces, "
+        "normals_to_vert_faces)\n"
+        "      USE constants_mod, ONLY: r_def, i_def\n"
+        "      IMPLICIT NONE\n"
+        "      INTEGER(KIND=i_def), intent(in) :: nlayers\n"
+        "      INTEGER(KIND=i_def), intent(in) :: ndf_w1\n"
+        "      INTEGER(KIND=i_def), intent(in), dimension(ndf_w1) :: map_w1\n"
+        "      INTEGER(KIND=i_def), intent(in) :: ndf_w2\n"
+        "      INTEGER(KIND=i_def), intent(in), dimension(ndf_w2) :: map_w2\n"
+        "      INTEGER(KIND=i_def), intent(in) :: ndf_w3\n"
+        "      INTEGER(KIND=i_def), intent(in), dimension(ndf_w3) :: map_w3\n"
+        "      INTEGER(KIND=i_def), intent(in) :: undf_w1, undf_w2, undf_w3\n"
+        "      REAL(KIND=r_def), intent(in) :: rscalar_1\n"
+        "      REAL(KIND=r_def), intent(inout), dimension(undf_w1) :: "
+        "field_2_w1\n"
+        "      REAL(KIND=r_def), intent(in), dimension(undf_w2) :: "
+        "field_3_w2\n"
+        "      REAL(KIND=r_def), intent(in), dimension(undf_w2) :: "
+        "field_4_w2\n"
+        "      REAL(KIND=r_def), intent(in), dimension(undf_w3) :: "
+        "field_5_w3\n"
+        "      INTEGER(KIND=i_def), intent(in) :: nfaces_re_h\n"
+        "      INTEGER(KIND=i_def), intent(in) :: nfaces_re_v\n"
+        "      REAL(KIND=r_def), intent(in), dimension(3,nfaces_re_h) :: "
+        "normals_to_horiz_faces\n"
+        "      REAL(KIND=r_def), intent(in), dimension(3,nfaces_re_v) :: "
+        "normals_to_vert_faces\n"
+        "    END SUBROUTINE testkern_ref_elem_code\n"
+        "  END MODULE testkern_ref_elem_mod")
+    assert output in gen
+
+
+def test_refelem_stub_arglist_err():
+    ''' Check that the KernStubArgList.ref_element_properties method raises
+    the expected error if it encounters an unsupported property. '''
+    from psyclone.psyGen import InternalError
+    from psyclone.dynamo0p3 import KernStubArgList
+    # Create the Kernel object
+    ast = fpapi.parse(os.path.join(BASE_PATH,
+                                   "testkern_ref_elem_all_faces_mod.F90"),
+                      ignore_comments=False)
+    metadata = DynKernMetadata(ast)
+    kernel = DynKern()
+    kernel.load_meta(metadata)
+    # Break the list of ref-element properties required by the Kernel
+    kernel.reference_element.properties.append("Wrong property")
+    with pytest.raises(InternalError) as err:
+        create_arg_list = KernStubArgList(kernel)
```

Comment:
Please could you have just the one statement within the `with` block so that it's clear which one generates the error.

Reply:
Done.